cmake/modules/kconfig.cmake

```diff
@@ -146,13 +147,19 @@ else()
   set(_local_TOOLCHAIN_HAS_PICOLIBC n)
 endif()
 
+# Extract git commit information for application
+git_commit_hash(${APPLICATION_SOURCE_DIR} GIT_COMMIT_HASH)
+git_commit_hash_short(${GIT_COMMIT_HASH} GIT_COMMIT_HASH_SHORT GIT_COMMIT_HASH_SHORT_INT)
```

Comment:
e.g. here, "from application" there is no guarantee that is correct

Reply:
Also what happens if the user is working on a freestanding application which is not part of a git repo, have you tested that behaviour here? You are always assuming that the git call will be successful

Reply:
>  e.g. here, "from application" there is no guarantee that is correct

I'm not sure how it could be incorrect since the function is provided with `${APPLICATION_SOURCE_DIR}`

Reply:
> Also what happens if the user is working on a freestanding application which is not part of a git repo, have you tested that behaviour here? You are always assuming that the git call will be successful

Fixed. I can't see any reasonable way to test the behaviour in CI, but the build now completes when either the first line is commented out or `unset(GIT_COMMIT_HASH)` is added between the two